go/config/config.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2019 The Grafeas Authors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+
+	fernet "github.com/fernet/fernet-go"
+	"gopkg.in/yaml.v2"
+)
+
+// File is the grafeas config file.
+type file struct {
+	Grafeas *config `yaml:"grafeas"`
+}
+
+type Config struct {
+	Address            string   `yaml:"address"`              // Endpoint address, e.g. localhost:8080 or unix:///var/run/grafeas.sock
+	CertFile           string   `yaml:"certfile"`             // A PEM eoncoded certificate file
+	KeyFile            string   `yaml:"keyfile"`              // A PEM encoded private key file
+	CAFile             string   `yaml:"cafile"`               // A PEM eoncoded CA's certificate file
+	CORSAllowedOrigins []string `yaml:"cors_allowed_origins"` // Permitted CORS origins.
+}
+
+type EmbeddedStoreConfig struct {
+	Path string `yaml:"path"` // Path is the folder path to storage files
+}
+
+type PgSQLConfig struct {
+	Host     string `yaml:"host"`
+	DbName   string `yaml:"dbname"`
+	User     string `yaml:"user"`
+	Password string `yaml:"password"`
+	// Valid sslmodes: disable, allow, prefer, require, verify-ca, verify-full.
+	// See https://www.postgresql.org/docs/current/static/libpq-connect.html for details
+	SSLMode       string `yaml:"sslmode"`
+	PaginationKey string `yaml:"paginationkey"`
+}
+
+// Config is the global configuration for an instance of Grafeas.
+type config struct {
+	API            *Config              `yaml:"api"`
+	StorageType    string               `yaml:"storage_type"` // Supported storage types are "memstore", "postgres" and "embedded"
+	PgSQLConfig    *PgSQLConfig         `yaml:"postgres"`
+	EmbeddedConfig *EmbeddedStoreConfig `yaml:"embedded"` // EmbeddedConfig is the embedded store config
+}
+
+// DefaultConfig is a configuration that can be used as a fallback value.
+func defaultConfig() *config {
+	return &config{
+		API: &Config{
+			Address:  "0.0.0.0:8080",
+			CertFile: "",
+			KeyFile:  "",
+			CAFile:   "",
+		},
+		StorageType: "memstore",
+		PgSQLConfig: &PgSQLConfig{},
+	}
+}
+
+// Creates a config from a YAML-file. If fileName is an empty
+// string a default config will be returned.
+func LoadConfig(fileName string) (*config, error) {
```

Comment:
Comments on exported methods should always start with the method name. E.g.,
// LoadConfig creates a ...

Reply:
good point, did a cleanup